basiclti/basiclti-impl/src/java/org/sakaiproject/lti/impl/LTIExportServiceImpl.java

```diff
@@ -0,0 +1,254 @@
+		}
+	}
+	
+	private String getCSVSeparator() {
+		if(csvSeparator == null) {
+			csvSeparator = serverConfigurationService.getString("basiclti.export.csv.separator", ",");
+		}
+		return csvSeparator;
+	}
+	
+	private void exportCSV(OutputStream out, String siteId, List<Map<String,Object>> contents) {
+		
+		StringBuilder result = new StringBuilder();
+		FormattedText validator = new FormattedText();
+		DateFormat dateFormatter = DateFormat.getDateTimeInstance(DateFormat.SHORT, DateFormat.SHORT, rb.getLocale());
+		
+		String attribution_name = serverConfigurationService.getString(LTIService.LTI_SITE_ATTRIBUTION_PROPERTY_NAME);
+		if(StringUtils.isNotEmpty(attribution_name)) {
+			//check if given property is a translation key
+			String aux = rb.getString(attribution_name);
+			if(StringUtils.isNotEmpty(aux)) 
+				attribution_name = aux;
+		}
+		
+		boolean isAdmin = isAdmin(siteId);
+		
+		//set header row
+		result.append(rb.getString("export.title"));
+		result.append(getCSVSeparator());
+		result.append(rb.getString("export.url"));
+		result.append(getCSVSeparator());
+		if(isAdmin) {
+			result.append(rb.getString("export.siteid"));
+			result.append(getCSVSeparator());
+			result.append(rb.getString("export.sitetitle"));
+			result.append(getCSVSeparator());
+		}
+		result.append(rb.getString("export.createdat"));
+		if(isAdmin) {
+			result.append(getCSVSeparator());
+			result.append(rb.getString("export.sitecontactname"));
+			result.append(getCSVSeparator());
+			result.append(rb.getString("export.sitecontactemail"));
+			if(StringUtils.isNotEmpty(attribution_name)) {
+				result.append(getCSVSeparator());
+				result.append(attribution_name);
+			}
+		}
+		result.append("\n");
+		
+		//values rows
+		for(Map<String,Object> content : contents) {
+			result.append(validator.escapeHtml((String)content.get("title"), false));
+			result.append(getCSVSeparator());
+			
+			String url = (String)content.get("launch");
+			if(StringUtils.isEmpty(url)) {
+				try {
+					url = (String)(ltiService.getToolDao(new Long(content.get(LTIService.LTI_TOOL_ID).toString()), siteId).get("launch"));
+				} catch(Exception e) {
+					url = "-";
+					e.printStackTrace();
+				}
+			}
+			result.append(validator.escapeHtml(url, false));
+			result.append(getCSVSeparator());
+			
+			if(isAdmin) {
+				result.append(validator.escapeHtml((String)content.get("SITE_ID"), false));
+				result.append(getCSVSeparator());
+				
+				result.append(validator.escapeHtml((String)content.get("SITE_TITLE"), false));
+				result.append(getCSVSeparator());
+			}
+						
+			try{ result.append(dateFormatter.format(content.get("created_at"))); } catch(Exception e){ result.append("-"); }
+			
+			if(isAdmin) {
+				result.append(getCSVSeparator());
+				result.append(validator.escapeHtml((String)content.get("SITE_CONTACT_NAME"), false));
+				result.append(getCSVSeparator());
+				
+				result.append(validator.escapeHtml((String)content.get("SITE_CONTACT_EMAIL"), false));
```

Comment:
When outputting CSVs you needs to escape special characters (comma) normally this is done by enclosing the field in double quotes and then escaping a double quote with an extra double quote.

Although this is simple there are small java libraries that will do this.